domain/secretbackend/type.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package secretbackend
+
+import (
+	"time"
+
+	"github.com/juju/worker/v4"
+
+	"github.com/juju/juju/core/changestream"
+	"github.com/juju/juju/core/secrets"
+	"github.com/juju/juju/core/watcher"
+)
+
+// CreateSecretBackendParams are used to create a secret backend.
+type CreateSecretBackendParams struct {
+	ID                  string
+	Name                string
+	BackendType         string
+	TokenRotateInterval *time.Duration
+	NextRotateTime      *time.Time
+	Config              map[string]interface{}
+}
+
+// UpdateSecretBackendParams are used to update a secret backend.
+type UpdateSecretBackendParams struct {
+	ID                  string
+	NameChange          *string
+	TokenRotateInterval *time.Duration
+	NextRotateTime      *time.Time
+	Config              map[string]interface{}
+}
+
+// SecretBackendFilter is used when listing secret backends.
+type SecretBackendFilter struct {
```

Comment:
We've decided to move away from filters in state. We can have a "ListSecretBackends" which takes a nillable list of names, which will be nil for All.
See `ListStoragePools(ctx context.Context, names domainstorage.Names...`